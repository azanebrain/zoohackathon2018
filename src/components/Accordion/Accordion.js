import styled from 'styled-components';

const Accordion = styled.button`
  background-color: ${props => (!props.isActive ? '#fff' : '#e7e7e8')};
  font-family: 'Raleway', sans-serif;
  font-weight: 600;
  font-size: 14px;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  text-align: left;
  border: none;
  outline: none;
  transition: 0.2s;
  position:relative;

  &::after {
    font-family: 'Lato', sans-serif;
    font-weight: 300;
    content: '${props => (!props.isActive ? '-' : '+')}';
    font-size: '${props => (!props.isActive ? '27px' : '42px')}';
    color: #777;
    float: right;
    margin-left: 5px;
    top: 50%;
    transform: translateY(-50%);
    position: absolute;
    right: 24px;
  }

  &::hover {
    background-color: #e7e7e8;
  }
`;

export default Accordion;
