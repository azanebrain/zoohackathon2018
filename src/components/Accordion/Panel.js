import styled from 'styled-components';

const Panel = styled.div`
background-color: white;
display: block;
max-height: ${props => (!props.isActive ? 0 : '500px')};
overflow: hidden;
padding: 0;
transition: max-height 0.25s linear;

.link-more { display: none; }
`;

const PanelContainer = styled.div`
padding: 0 16px 16px 84px;
font-size: 14px;
`;

export { Panel, PanelContainer };
