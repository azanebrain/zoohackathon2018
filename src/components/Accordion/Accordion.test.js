import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Accordion from './Accordion';

test('Active Accordion renders with background color', () => {
  const tree = renderer.create(<Accordion isActive={true} />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('background-color', '#e7e7e8');
})

test('Inactive Accordion renders with background color', () => {
  const tree = renderer.create(<Accordion isActive={false} />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('background-color', '#fff');
})