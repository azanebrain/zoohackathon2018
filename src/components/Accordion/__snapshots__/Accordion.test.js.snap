// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Active Accordion renders with background color 1`] = `
.c0 {
  background-color: #e7e7e8;
  font-family: 'Raleway',sans-serif;
  font-weight: 600;
  font-size: 14px;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  text-align: left;
  border: none;
  outline: none;
  -webkit-transition: 0.2s;
  transition: 0.2s;
  position: relative;
}

.c0::after {
  font-family: 'Lato',sans-serif;
  font-weight: 300;
  content: '+';
  font-size: '42px';
  color: #777;
  float: right;
  margin-left: 5px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  position: absolute;
  right: 24px;
}

.c0::hover {
  background-color: #e7e7e8;
}

<button
  className="c0"
/>
`;

exports[`Inactive Accordion renders with background color 1`] = `
.c0 {
  background-color: #fff;
  font-family: 'Raleway',sans-serif;
  font-weight: 600;
  font-size: 14px;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  text-align: left;
  border: none;
  outline: none;
  -webkit-transition: 0.2s;
  transition: 0.2s;
  position: relative;
}

.c0::after {
  font-family: 'Lato',sans-serif;
  font-weight: 300;
  content: '-';
  font-size: '27px';
  color: #777;
  float: right;
  margin-left: 5px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  position: absolute;
  right: 24px;
}

.c0::hover {
  background-color: #e7e7e8;
}

<button
  className="c0"
/>
`;
